students/KyleCreek/lesson01/assignment/test_integration.py

```diff
@@ -0,0 +1,167 @@
+        with unittest.mock.patch('builtins.input', return_value='1'):
+            self.assertEqual(Main.main_menu(), Main.add_new_item)
+        with unittest.mock.patch('builtins.input', return_value='2'):
+            self.assertEqual(Main.main_menu(), Main.item_info)
+        with unittest.mock.patch('builtins.input', return_value='q'):
+            self.assertEqual(Main.main_menu(), Main.exit_program)
+
+    def test_exit_program(self):
+        """
+        Tests the Exit Program functionality of Main.py
+        Ensures function will exit program when called.
+        :return: None
+        """
+
+        # Ensures System Exit Is Raised WHen Program is Exited.
+        with self.assertRaises(SystemExit):
+            Main.exit_program()
+
+    def test_get_price(self):
+        """
+        Tests get Price Method of Main.py
+        Ensures the price will be returned when called
+        :return: None
+        """
+        # Create patch to handle each instance of print
+        with unittest.mock.patch('sys.stdout', new=io.StringIO()) as fake_out:
+            Main.get_price()
+
+        # Obtain StringIO output from get price
+        fake_out = fake_out.getvalue()
+        fake_out = fake_out.strip()
+        # Verify Main.get_price() produces desired output
+        self.assertEqual(fake_out, "Get price")
+
+    def test_add_new_item(self):
+        """
+        Tests the add_new_item method of Main.py. Ensures
+        correct objects are instantiated based
+        when provided simulated input.
+        :return: None
+        """
+        # Create a list of inputs as 'side_effects' to handle
+        # each instances of 'input()' function in Main.py
+        mock_input1 = [1, "Couch", 100, "y", "Plastic", "L"]
+
+        # Create to handle input function of Main.py and pass desired
+        # Inputs
+        with unittest.mock.patch('builtins.input', side_effect=mock_input1):
+            # Create a new item instance based on mock_input1 inputs
+            Main.add_new_item()
+            self.assertIn(1, Main.FULL_INVENTORY)
+            # Extract the dictionary created in Main.py
+            new_dict = Main.FULL_INVENTORY[1]
+            # Assert Values have been properly added to objects
+            self.assertEqual("Couch", new_dict['description'])
+            self.assertEqual(100, new_dict['rental_price'])
+            self.assertEqual("Plastic", new_dict['material'])
+            self.assertEqual("L", new_dict['size'])
+
+        # Create a list of inputs as 'side_effects' to handle
+        # each instances of 'input()' function in Main.py
+        mock_input2 = [2, "Blender", 100, "n", 'y', "Sony", 100]
+
+        # Create to handle input function of Main.py and pass desired
+        # Inputs
+        with unittest.mock.patch('builtins.input', side_effect=mock_input2):
+            # Create a new item instance based on mock_input2 inputs
+            Main.add_new_item()
+            self.assertIn(2, Main.FULL_INVENTORY)
```

Comment:
The idea of your test is: 1) Create a record to be inserted; 2) Mock input from the record (list); 3) Check whether the record with particular ID is in FULL_INVENTORY; 3) Check whether the values are completely inserted